doc/en/user/source/community/oauth2/index.rst

```diff
@@ -0,0 +1,151 @@
+------------------------------------------------
+
+This module allows GeoServer to authenticate against the `OAuth2 Protocol <https://tools.ietf.org/html/rfc6749>`_.
+
+In order to let the module work, it's mandatory to setup and configure both ``oauth2`` and ``oauth2-xxxx-extension``.
+
+The first one contains the necessary dependencies of the OAuth2 core module. This module contains the implementation of the 
+GeoServer security filter, the base classes for the OAuth2 Token services and the GeoServer GUI panel.
+
+Since in almost all cases the only thing different between OAuth2 Providers are the endpoint URIs and the client connection
+information (not only the keys - public and secret - but also the user profile representations), in order to allow GeoServer
+connecting to a specific OAuth2 provider it is sufficient to install the OAuth2 Core module plugin (and correctly configure
+the parameters through the GeoServer GUI - see next section for the details) and the concrete implementation of the OAuth2
+REST token template and resource details.
+
+Currently this module is shipped with a sample extension for Google OAuth2 Provider. This is a particular case since the 
+Google JWT response is not standard and therefore we had to define and inject also a ``GoogleUserAuthenticationConverter`` taking
+the Google REST response against a valid ``access_token`` and converting it to a OAuth2 standard one.
+
+Other than this the most interesting part is the implementation of the base class ``GeoServerOAuth2SecurityConfiguration``.
+
+The latter contains the Google implementation of the ``OAuth2RestTemplate``.
+
+In the next section we will see how to install and configure the OAuth2 security filter on GeoServer authenticating against 
+Google OAuth2 Provider.
+
+Configure the Google authentication provider
+--------------------------------------------
+
+The first thing to do is to configure the OAuth2 Provider and obtain ``Client ID`` and ``Client Sectet`` keys.
+
+#. Obtain OAuth 2.0 credentials from the Google API Console.
+
+   Visit the `Google API Console <https://console.developers.google.com/>`_ to obtain OAuth 2.0 credentials such as a client ID and client secret 
+   that are known to both Google and your application. The set of values varies based on what type of application you are building. 
+   For example, a JavaScript application does not require a secret, but a web server application does.
+   
+    .. figure:: images/google1.jpg
+       :align: center
+
+#. Create a new ``OAuth client ID``
+
+   Click on the ``Client credentials`` context menu as shown in the figure below.
+   
+    .. figure:: images/google2.jpg
+       :align: center
+
+#. Configure a new ``Web application``
+
+   Add the ``Name`` and the ``Authorized redirect URIs`` like shown here below.
+   
+    .. note:: Always add two entries for each URI. One without the ending ``/`` and another one with it.
+   
+    .. figure:: images/google3.jpg
```

Comment:
about this specific issue... in this case Google does not accept the URI because you typed "locahost" instead of "localhost".

Google does checks on the address. It cannot be a public IP or a local domain different from localhost (like http://mydomain.local for instance).